### ISSUE TITLE
Support for Multiple Streams over Single Peer Connection

### ISSUE THREAD

**Issue by peterzanetti** [ant-media/Ant-Media-Server#2937]
**Describe the solution you'd like**
Be able to establish a single peer connection to the media server, and received multiple playback streams.

**Describe alternatives you've considered**
The alternative is the current design, 1 peer connection per stream.

**Additional context**
Lots of peer connections is a fragile method and prone to failure. In order to scale larger meetings (really anything above 10), adding more peer connections adds complexity and overhead (both cpu and network) that the browser needs to contend with. Other media servers such as jitsi and mediasoup both have single peer connection options out of the box.

**Comment by mekya** [comment]
Yes you're right @peterzanetti.  We put it on our roadmap. Just let us schedule it.

**Comment by peterzanetti** [comment]
Any update on this?

**Comment by mekya** [comment]
It's still under development. We encountered some problems in implementation and trying to resolve them.

**Comment by ducle** [comment]
7 months. Could you update on this please?

**Comment by mekya** [comment]
Hi @ayazhussein ,

Thank you for the follow up.  This feature is currently available for the playback side.
We plan to support multiple tracks on video publish as well.

**Comment by mekya** [comment]
Hi @ayazhussein ,
Thank you.  There is a blogpost 
https://antmedia.io/antmediaserver-webrtc-multitrack-playing-feature/

My colleagues will check if it's up to date and provide the documentation. 

Please let me know if you stuck somewhere and I can help with anything

Thank you

**Comment by ayazhussein** [comment]
Hello @mekya,

Yes, I don't see where I define the groupId, when I played with the multitrack html file, I couldn't view the main stream, only the subtracks after adding them manually.

Is there a way to set a group of streams so that on the UI i can just get all the streams by the groupId?

My use case is showing IP cameras (using stream source) from different locations and It would be nice if I could group them and start/stop them by groupId or at least start the stream via WebRTC by the groupId.

Thanks,

**Comment by mekya** [comment]
Hi @ayazhussein ,

I think there is just a simple trick there. Let me check and get back to you

Regards,
A. Oguz

**Comment by mekya** [comment]
Hi @ayazhussein ,

I've just had a chance to take a look.

Let me tell the design behind this approach and provide you a solution

Firstly, group id or main track is actually not a real stream, it's just a logical stream. I think this is why it is disabled to play the main track in the multitrackplayer.html. You can enable the maintrack player by just commenting out these lines 

```
		if(index == room) {
			return;
		}
```
at
https://github.com/ant-media/StreamApp/blob/master/src/main/webapp/multitrackplayer.html#L296 

My recommendation is keeping the maintrack as logical stream not a real stream. 

Btw, I've found an issue in the main track playback, create an issue https://github.com/ant-media/Ant-Media-Server/issues/4224 and send the fix. You can get monitor the latest snapshot here -> https://antmedia.io/changelog/changelog.html

It will likely be available later today and you'll see the reference of this issue in the changelog(https://github.com/ant-media/Ant-Media-Server/issues/4224)
